### ISSUE TITLE
Store application-specific data on tuning runs and trials

### ISSUE THREAD

**Issue by jbosboom** [jansel/opentuner#59]
The mario demo has pluggable representations.  When doing comparative experiments, it would be convenient to record the name of the representation used in the database.

Mario would also like to record the raw values it gets back from each trial (win/loss, pixels moved right, elapsed frames) alongside the fitness function, which will help interpret the effects of fitness functions like "progress times average speed" whose values are not obvious when displayed on a graph.

**Comment by jansel** [comment]
Does #58 fix this?

**Comment by jbosboom** [comment]
I don't think so.  I mean that the Mario code wants to a) record it used NaiveRepresentation/DurationRepresentation/etc and Progress/ProgressTimesAverageSpeed/etc and b) record the actual progress and elapsed frames for each trial.  OpenTuner doesn't need this information for anything -- it's for the humans developing a particular application that uses OpenTuner.

#58 includes some machine-readable dump of the parameters, but I want a human-meaningful value so I can kick off 20 runs of each representation on the cluster and be able to nicely graph the results later.

#58 doesn't do any trial-specific recording, at least that I can see.

**Comment by Sierangho** [comment]
Recording a human readable name for representations might be difficult
without building the notion of a representation into OpenTuner. One
workaround could be to set the version in ProgramVersion to the
representation name? It might be worth adding a representation-name column
to ProgramVersion though (it kind of matches up with the machine-readable
parameter dump).

Similarly, instead of making a new Mario-specific table for recording
actual progress/elapsed frames in each result, you could fake it and store
them under "energy" and "size" in the Result or something.

**Comment by jbosboom** [comment]
I'd be happy just adding a column for an application-defined JSON string (or whatever the application wants to put in there) on tuning runs and results.

Alternately, we could decide the database isn't the right place for this and applications should just keep their own logs in whatever format they desire.
